pkg/server/server.go

```diff
@@ -1372,14 +1517,57 @@ func (s *Server) ListStores(ctx context.Context, req *openfgav1.ListStoresReques
 
 	ctx = telemetry.ContextWithRPCInfo(ctx, telemetry.RPCInfo{
 		Service: s.serviceName,
-		Method:  "ListStores",
+		Method:  authz.ListStores,
 	})
 
+	stores, err := s.CheckAuthzListStores(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	q := commands.NewListStoresQuery(s.datastore,
 		commands.WithListStoresQueryLogger(s.logger),
 		commands.WithListStoresQueryEncoder(s.encoder),
 	)
-	return q.Execute(ctx, req)
+
+	if s.authorizer == nil {
+		return q.Execute(ctx, req)
+	}
+
+	storesMap := make(map[string]struct{})
+	re := regexp.MustCompile(`^.*:(.*)`)
+	for _, store := range stores {
+		id := re.FindStringSubmatch(store)
+		storesMap[id[1]] = struct{}{}
+	}
+
+	var accessibleStores []*openfgav1.Store
+	var continuationToken string
+
+	for {
+		req.ContinuationToken = continuationToken
+		resp, err := q.Execute(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, store := range resp.GetStores() {
+			if _, ok := storesMap[store.GetId()]; ok {
+				accessibleStores = append(accessibleStores, store)
+			}
+		}
+
+		continuationToken = resp.GetContinuationToken()
+
+		if len(accessibleStores) > 0 || resp.GetContinuationToken() == "" {
+			break
+		}
+	}
+
+	return &openfgav1.ListStoresResponse{
+		Stores:            accessibleStores,
+		ContinuationToken: continuationToken,
+	}, nil
```

Comment:
I don't understand what or why this code is doing is doing what it's doing. You are paginating through all accessible stores but also returning a token...?

Reply:
This will need to change before the work is done - maybe we should consider taking ListStores out into it's own PR and keeping it out of this particular PR

Reply:
Took ListStores out for now.